New structure

artem-melnikov opened the pull request.

dmitry-prohorov committed:
Update README.md

artem-melnikov committed:
Merge pull request #1 from dmitry-prohorov/patch-1

Update README.md

artem-melnikov committed:
added image

artem-melnikov committed:
Merge remote-tracking branch 'origin/master'

artem-melnikov committed:
новая структура серверной стороны

artem-melnikov committed:
обновлена клиентская часть

artem-melnikov committed:
добавлено отображение кастомных объектов

artem-melnikov committed:
добавил flsChecker

artem-melnikov committed:
немного отрефакторил код

artem-melnikov committed:
added classes: AccessibleObjectSetting, DynamicQuery, ObjectDescription

dmitry-prohorov commented on src/classes/ObjectDescription.cls:
```diff
@@ -0,0 +1,44 @@
+    }
+    public DescribeFieldResult getFieldDescribe(String fieldName) {
+        DescribeFieldResult fieldDescription = this.fieldsDescribe.get(fieldName).getDescribe();
+        if (fieldDescription == null) {
```
`if null return null`?

artem-melnikov committed:
added classes: ObjectSetting, ResponseWrapper
changed classes: DynamicQuery, ObjectDescription, StatusComponentService

dmitry-prohorov commented on src/classes/DynamicQuery.cls:
```diff
@@ -5,64 +5,45 @@
+
         //checking access to object
-        if (!settings.isAccessible()) {
+        ObjectDescription currentObjectDescription = new ObjectDescription(settings.objectType);
```
ObjectDescription should be singleton

dmitry-prohorov commented on src/classes/DynamicQuery.cls:
```diff
@@ -5,64 +5,45 @@
-                return '';
-            }
-            SObject record = records.get(0);
+        String relatedRecordId = (String) recordsList.get(0).get(settings.lookupField);
```
recordsList potential NPE

dmitry-prohorov commented on src/classes/DynamicQuery.cls:
```diff
@@ -5,64 +5,45 @@
+        String queryString = 'select ' + settings.title;
+        //adding fields to which have access
+        for (String field : settings.fields) {
+            if (!currentObjectDescription.getFieldDescribe(field).isAccessible()) {
```
dynamic query class should not check accessible access for fields

dmitry-prohorov commented on src/classes/ResponseWrapper.cls:
```diff
@@ -0,0 +1,23 @@
+            if (record.get(field) == null) {
+                continue;
+            }
+            String fieldLabel = ObjectDescription.ObjectsDescriptionMap.get(settings.objectType).getFieldDescribe(field).getLabel();
```
`ObjectsDescriptionMap` variable name should start from UpperCase

dmitry-prohorov replied:
you should not have access to `ObjectsDescriptionMap` outside of `ObjectDescription` class

dmitry-prohorov commented on src/classes/StatusComponentService.cls:
```diff
@@ -12,92 +12,30 @@ public with sharing class StatusComponentService {
     }
 
-    private Object getObjectsJSON(List<String> jsonSettings, String Id) {
+    private Object getResponse(List<String> jsonSettings, String Id) {
```
not informative method name

dmitry-prohorov commented on src/classes/ResponseWrapper.cls:
```diff
@@ -0,0 +1,23 @@
+ * Created by User on 12.10.2017.
+ */
+
+public with sharing class ResponseWrapper {
```
not very informative name of wrapper

dmitry-prohorov replied:
it is better to use wrapper postfix for standard objects, custom wrapper should not contain this postfix

dmitry-prohorov commented on src/classes/ObjectSetting.cls:
```diff
@@ -0,0 +1,24 @@
+    };
+
+    //constructor
+    public ObjectSetting(String ot, String cot, String lf, String t, String i, List<String> f) {
```
object setting is a wrapper around metadata object and should take as parameter __mdt object

artem-melnikov committed:
added classes: AccessChecker
changed classes: ObjectSetting, DynamicQuery, ObjectDescription, StatusComponentService, ResponseWrapper to StatusComponentResponse

artem-melnikov committed:
added classes: AggregationResponse, AggregationSetting
changed classes: StatusComponentService

artem-melnikov committed:
refactored some code
